src/sentry/api/endpoints/relay/project_configs.py

```diff
@@ -67,7 +67,33 @@ def _post(self, request: Request):
                 full_config_requested=full_config_requested,
             )
         else:
-            return Response("Unsupported version, we only support versions 1 to 3.", 400)
+            return Response("Unsupported version, we only support versions 1 to 4.", 400)
+
+    def _get_v4_config(self, request):
+        res = self._post_or_schedule_by_key(request)
+        metrics.incr("relay.project_configs.post_v4.pending", amount=len(res["pending"]))
+        metrics.incr("relay.project_configs.post_v4.fetched", amount=len(res["configs"]))
```

Comment:
In the current state, it's possible a request comes in with no projects but requesting the global config. In this case, emitting a counter metric with `amount=0` doesn't provide any observability insights but still creates the overhead of generating the metric. What do you think if we only emit these when `amount>0`?

Reply:
Related -- is there any observability we can add that could help us get more insights around global configs, the new functionality in `v4`?

Reply:
sounds good to me, I suppose we would wanna do it on v3 too?

for the 2nd, I suppose you mean adding a similar metric to global right? that makes a lot of sense to me

Reply:
now we won't request project configs along with globalconfig anymore, as we request globalconfig totally separately, so then i dont think we need to change this